compiler/rustc_infer/src/infer/error_reporting/need_type_info.rs

```diff
@@ -363,6 +119,64 @@ impl UnderspecifiedArgKind {
     }
 }
 
+fn ty_to_string<'tcx>(infcx: &InferCtxt<'_, 'tcx>, ty: Ty<'tcx>) -> String {
+    let mut s = String::new();
+    let mut printer = ty::print::FmtPrinter::new(infcx.tcx, &mut s, Namespace::TypeNS);
+    let ty_getter = move |ty_vid| {
```

Comment:
Given these, we could potentially also return the list of type params to use them as part of the message.

Reply:
Comment still applies :)

Reply:
added this to #94483